script/commands/deploy.ts

```
@@ -0,0 +1,190 @@
+    (await deployContract(ctx, client, "hpl_validator_announce", {
+      hrp,
+      mailbox: deployment.mailbox.address,
+    }));
+  if (preload?.validator_announce)
+    log(`${deployment.validator_announce.type} already deployed`);
+
+  return deployment;
+};
+
+const deployIsms = async (
+  ctx: Context,
+  client: Client
+): Promise<ContextDeployments["isms"]> => {
+  if (!config.deploy.ism) {
+    throw new Error("ISM deployment config not found");
+  }
+
+  const log = (v: string) => console.log("[ism]".green, v);
+  const preload = ctx.deployments.isms;
+  const deployment =
+    preload ||
+    (await deployIsm(
+      ctx,
+      client,
+      config.deploy.ism instanceof Array
+        ? {
+            // default ism (multisig)
+            type: "multisig",
+            owner: client.signer,
+            validators: config.deploy.ism
+              .map((domain) => ({
+                domain,
+                addrs: [client.signer_addr],
+                threshold: 1,
+              }))
+              .reduce(
+                (acc, v) => ({
+                  [v.domain]: { addrs: v.addrs, threshold: v.threshold },
+                  ...acc,
+                }),
+                {} as Record<string, any>
+              ),
+          }
+        : config.deploy.ism
+    ));
+  if (preload) {
+    log(`ism ${deployment.type} already deployed`);
+  }
+
+  return deployment;
+};
+
+const deployHooks = async (
+  { networkId }: { networkId: string },
+  ctx: Context,
+  client: Client
+): Promise<ContextDeployments["hooks"]> => {
+  if (!config.deploy.hooks) {
+    throw new Error("Hook deployment config not found");
+  }
+
+  const log = (v: string) => console.log("[hooks]".green, v);
+  const preload = ctx.deployments?.hooks;
+  const deployment = preload || {};
+
+  deployment.default =
+    preload?.default ||
+    (await deployHook(
+      networkId,
+      ctx,
+      client,
+      config.deploy.hooks.default || { type: "mock" }
```

Let's not default to mock here, but throw an error, too hard to debug if you forgot to set the type

---

Agreed, let's get rid of this.